pkg/scheduler/queue/queue.go

```diff
@@ -282,35 +266,43 @@ func (q *RequestQueue) enqueueRequestToBroker(broker *queueBroker, r requestToEn
 	return nil
 }
 
-// tryDispatchRequestToQuerier finds and forwards a request to a waiting GetNextRequestForQuerier call, if a suitable request is available.
-// Returns true if call should be removed from the list of waiting calls (eg. because a request has been forwarded to it), false otherwise.
-func (q *RequestQueue) tryDispatchRequestToQuerier(broker *queueBroker, call *nextRequestForQuerierCall) bool {
-	req, tenant, idx, err := broker.dequeueRequestForQuerier(call.lastUserIndex.last, call.querierID)
+// trySendNextRequestForQuerier finds and forwards a request to a waiting GetNextRequestForQuerier call.
+//
+// Returns true if a nextRequestForQuerier was written to the nextRequestForQuerierCall's result channel.
```

Comment:
[nit] Might be good to clarify the implication of this, something like:

```suggestion
// Returns true if a nextRequestForQuerier was written to the nextRequestForQuerierCall's result channel, and so should be removed from the queue.
```